px-map-tile-layer-google.html

```diff
@@ -0,0 +1,140 @@
+    </px-map>
+
+@element px-map-tile-layer-google
+@blurb Loads base layer tiles from the Google Maps API
+@homepage index.html
+@demo index.html
+-->
+
+<dom-module id="px-map-tile-layer-google">
+  <style>
+    :host { display: none }
+  </style>
+  <template></template>
+</dom-module>
+<script>
+  Polymer({
+    is: 'px-map-tile-layer-google',
+    behaviors: [PxMapBehavior.Layer],
+
+    properties: {
+      /**
+       * A valid Google Maps Tile API key. Required to fetch the tiles - if it is not
+       * provided, no tiles will be loaded.
+       */
+      key: {
+        type: String,
+        observer: 'shouldUpdateInst'
+      },
+
+      /**
+       * The map type to use. Choose from the following options:
+       *
+       * - 'roadmap' - (default) The standard Google Maps painted map tiles.
+       * - 'satellite' - Satellite imagery.
+       *
+       * The following Google imagery sets are not supported: 'terrain, streetview'
+       */
+      imagery: {
+        type: String,
+        value: 'roadmap',
+        observer: 'shouldUpdateInst'
+      },
+
+      /**
+       * A region identifier representing the physical location of the user to whom
+       * the tiles are shown.
+       * Accepts Unicode region subtag identifiers which (generally) have a one-to-one
+       * mapping to country code Top-Level Domains (ccTLDs).
+       */
+      region: {
+        type: String,
+        value: 'us',
+        observer: 'shouldUpdateInst'
+      },
+
+      /**
+       * A Common Locale Data Repository language identifier indicating the
+       * language information on the tiles should be presented in.
+       */
+      language: {
+        type: String,
+        value: 'en-US',
+        observer: 'shouldUpdateInst'
+      },
+
+      /**
+       * Optional custom styling for the map.
+       * To learn more about map stying options, see:
+       * https://developers.google.com/maps/documentation/javascript/styling.
+       */
+      mapStyle: {
+        type: Array,
+        value: [],
+        observer: 'shouldUpdateInst'
+      }
+    },
+
+    canAddInst: function() {
+      return (typeof this.key === 'string') && this.key.length;
+    },
+
+    createInst: function(options) {
+      return L.tileLayer.google(options);
+    },
+
+    updateInst: function(lastOptions, nextOptions) {
+      if (lastOptions.GoogleTileAPIKey !== nextOptions.GoogleTileAPIKey) {
+        this.elementInst.setKey(nextOptions.GoogleTileAPIKey);
+      }
+      if (lastOptions.imagery !== nextOptions.imagery) {
+        this.elementInst.setMapType(nextOptions.imagery);
+      }
+      if (lastOptions.language !== nextOptions.language) {
+        this.elementInst.setLanguage(nextOptions.language);
+      }
+      if (lastOptions.region !== nextOptions.region) {
+        this.elementInst.setRegion(nextOptions.region);
+      }
+      if (lastOptions.mapStyle !== nextOptions.mapStyle) {
```

Comment:
Because `mapStyle` is an Array, if it is changed by the developer by mutating its members but not changing its reference this equality change might fail.

```
var arr = ['a', 'b', 'c'];
var arrReference = arr;
arr.push('d');
console.log(arrReference === arr) //=> true
```

This can make it hard for developers to dynamically update the mapStyle property in a way that triggers your `this.elementInst.setMapStyle` function and changes the mapStyles in the layer.

In px-map in other places we have tried to be nice and hash Object/Array properties into a string so we can do a deep equality check.

For an example, see the `featureStyle` property in px-map-layer-geojson:

https://github.com/PredixDev/px-map/blob/e122b4817c9575177b69fa0fde24e649658c7395/px-map-behavior-layer-geojson.es6.js#L227-L235

In `getInstOptions` we create a `featureStyleHash` using JSON.stringify. Then in `updateInst` we compare the last hash against the new hash to decide if we should do an update. (There's some other trickery to do with checking the featureStyle keys length in px-map-layer-geojson#updateInst... I don't think we need to do that here, we can probably just do a strict equality check on the last hash string vs. the next hash string.) 

Perhaps we could do the same thing here to ensure this updates? If we went that route we could do this here:

```
  if (lastOptions.mapStyleHash !== nextOptions.mapStyleHash) {
    ...
  }
```